### PR TITLE
Implemented 2 pki authentication controls

### PR DESCRIPTION
**Pull request by cdefrates**

### CODE REVIEW

File: ansible/roles/stig/tasks/TOSS_04_010020.yml [hunk 1]
```diff
@@ -10,7 +10,29 @@
 
 
 - name: TOSS-04-010020 - TOSS, for PKI-based authentication, must enforce authorized access to the corresponding private key.
-  # TODO: Implement the task
+  block:
+    - name: Find all private ssh key files
+      ansible.builtin.command: 'find / -xdev -type f -regex ".*/id_[a-z0-9]*[^\.pub]$"'
```
Comment (IanLee1521) [inline ansible/roles/stig/tasks/TOSS_04_010020.yml:15]: I didn't actually know about `-regex`, that's cool! -- I've always done variations on `-name`

File: ansible/roles/stig/tasks/TOSS_04_010020.yml [hunk 2]
```diff
@@ -10,7 +10,29 @@
...
+      register: passphrase_prompts
+      changed_when: false
+      failed_when: false
+      loop: '{{ nouser_files.stdout_lines }}'
+    - name: Verify all the ssh keys prompted for a passphrase
+      ansible.builtin.assert:
+        that:
+          "'Enter passphrase' in item.stdout"
+        success_msg: "The private ssh key has a passphrase"
+        fail_msg: "A private ssh key is does not have a passphrase set. Set it with ssh-keygen -n [passphrase]"
```
Comment (IanLee1521) [inline ansible/roles/stig/tasks/TOSS_04_010020.yml:34]: Is there any way to report which ssh key(s) doesn't have the passphrase ?

File: ansible/roles/stig/tasks/TOSS_04_010020.yml [hunk 3]
```diff
@@ -23,3 +45,4 @@
     - CCI-000186
     - isso_documented
     - pki_authentication
+    - manual
```
Comment (IanLee1521) [inline ansible/roles/stig/tasks/TOSS_04_010020.yml:48]:
```suggestion
    - manual
    - hpc_issue
```

This is one that's going to be tricky for us generally / has some concerns, so let's add that tag

**Review by IanLee1521 — APPROVED** [review]
I think we should add the hpc_issue tag (suggestion inline that you can apply from there since I'm actually logged in this time).

File: ansible/roles/stig/tasks/TOSS_04_010020.yml (hunk 2 above)
Comment (cdefrates) [reply ansible/roles/stig/tasks/TOSS_04_010020.yml:34]: Right now this test runs for each individual key, so you will get a block of details(including the file path) followed by the success/fail message for each key, so you can tell which ones need to be fixed.

**Pull request merged**